spec/template_deployment_service_spec.rb

```diff
@@ -8,11 +8,9 @@
 describe "TemplateDeploymentService" do
   before do
     setup_params
-
-    class String
-      def body; self; end
-      def headers; {}; end
-    end
+    allow_any_instance_of(String).to receive(:body).and_return(self)
```

Comment:
returning `self` caused the spec test failure because `self` is `RSpec::ExampleGroups::...InstanceMethods`. A string is expected from the `body` method.